### ISSUE TITLE
Bash shell parameter expansions of the form ${parameter@operator} cause a parsing error in osh

### ISSUE THREAD

**Issue by Ian2020** [oilshell/oil#1119]
Running the following valid bash script with `osh`:

```bash
blah="hello" ; echo "${blah@u}"
```

Will give:

```
oiltest:3: Unexpected token in ${} (VOp3)
```

These kind of parameter@operator expansions appear to have been introduced in BASH 4.4 according to the release announcement [Bash-4.4 Release available](https://lists.gnu.org/archive/html/bug-bash/2016-09/msg00018.html) - see point 1.(o). However I am testing on BASH 5.1.

They are described in full at the bottom of [Shell Parameter Expansion (Bash Reference Manual)](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion).

**Comment by andychu** [comment]
Thanks for the report!   I knew about `${x@Q}` which was introduced in 4.4, but it looks like they added `u U l L` in  bash 5.1.

Which is weird because there is already `${x,}` and `${x,,}` for lowercase, and `${x^}` and `${x^^}` for uppercase.

However those have weird caveats with globs, and also I think `,` and `^` are harder to remember.

I added a failing test case for this:

https://github.com/oilshell/oil/commit/cde8aac39523c9c04bbcb805b21a4d48c8fa8ea8

but we need to do a couple things:

1. upgrade the spec tests to bash 5.1 so we can properly test that it matches
2. figure out any locale stuff?    Hm I guess I could add test cases for those now ...

**Comment by andychu** [comment]
Yeah I was reminded that we not handling unicode characters in case folding, so I added some test cases there too ...

**Comment by Ian2020** [comment]
Thanks for this - I am using simple `a` -> `A` conversion nothing fancy.
